### PR TITLE
Add BS 200-299 or BS >=300 graph

### PR DESCRIPTION
**Pull request by PriyangaPKini**
**Story card:** [sc-7781](https://app.shortcut.com/simpledotorg/story/7781/add-bs-200-299-or-bs-300-graph)

## Because

We need add a graph showing the trend of BS 200-299 or BS >=300 patients on the diabetes reports page

## This addresses

- Region Summary and Region Summary Schema report blood sugar metrics
- Repository and Repository presenter present data to the controller
- The diabetes view adds html to display the BS 200-299 or BS >=300 graph

## Test instructions

1. Turn on `diabetes_management_reports` on flipper
2. Visit the reports page of a facility which has `enable_diabetes_management` set to true
3. Click on `Diabetes` link
4. You should see the BS 200-299 or BS >=300 graph
Screenshot 2022-05-04 at 6 32 27 PM

 Screenshot 2022-05-04 at 6 30 21 PM

### CODE REVIEW

File: app/assets/javascripts/common/reports.js [hunk 1]
```diff
@@ -883,15 +886,156 @@ Reports = function (withLtfu) {
...
-    if (bsBelow200GraphCanvar) {
-      new Chart(bsBelow200GraphCanvar.getContext("2d"), bsBelow200GraphConfig);
+    if (bsBelow200GraphCanvas) {
+      new Chart(bsBelow200GraphCanvas.getContext("2d"), bsBelow200GraphConfig);
       populateBSBelow200GraphDefault();
     }
   };
 
+  this.setupBSOver200Graph = (data) => {
+      console.log(data);
```
Comment (prabhanshuguptagit) [inline app/assets/javascripts/common/reports.js:899]:
```suggestion

```

File: app/models/reports/region_summary_schema.rb [hunk 2]
```diff
@@ -173,10 +173,10 @@ def initialize(regions, periods:)
...
         group_field = case group_by
-          when /user_id\z/ then :user_id
-          when /gender\z/ then :patient_gender
-          when nil then nil
-          else raise(ArgumentError, "unknown group for follow ups #{group_by}")
+                      when /user_id\z/ then :user_id
+                      when /gender\z/ then :patient_gender
+                      when nil then nil
+                      else raise(ArgumentError, "unknown group for follow ups #{group_by}")
         end
```
Comment (prabhanshuguptagit) [inline app/models/reports/region_summary_schema.rb:180]: 👾

**Review by prabhanshuguptagit — APPROVED** [review]
Looks great ✨

**Comment by prabhanshuguptagit** [comment]
Found a small bug: Clicking on the LTFU button on this card toggles the toggle in the card above.

**Comment by PriyangaPKini** [comment]
> Found a small bug: Clicking on the LTFU button on this card toggles the toggle in the card above.

We have already documented this bug in another story. This bug is also there on the overview page as well. Will fix in a separate PR

**Comment by prabhanshuguptagit** [comment]
> We have already documented this bug in another story

The LTFU toggles toggling together is intentional, not a bug. I was referring to this animation 
https://user-images.githubusercontent.com/16774200/166898976-efac2587-9a75-4f4f-92d9-1d88d8e1df52.mov

Should fix if you change the id [here](https://github.com/simpledotorg/simple-server/pull/3708/files#diff-70d327246c197a5af69465e48d3e01adc194e34eed0ac7ef6e51edef4bfea07bR153).

**Pull request merged**